src/Lease/Lease.Aggregate.fs

```diff
@@ -8,57 +8,17 @@ type ObservationDate =
     | AsOf of DateTime
     | AsAt of DateTime
 
-type EvolveDomain<'DomainEvent,'DomainState> =
-    'DomainState
-     -> 'DomainEvent
-     -> 'DomainState
+type EffectiveLeaseEvents = LeaseEvent list
 
-type InterpretDomain<'DomainCommand,'DomainEvent,'DomainState> =
-    'DomainCommand
-     -> 'DomainState
-     -> Result<'DomainEvent list,string>
-
-type EffectiveEvents<'DomainEvent> = 'DomainEvent list
-
-type Reconstitute<'DomainEvent,'DomainState> =
-    ObservationDate
-     -> EffectiveEvents<'DomainEvent>
-     -> 'DomainState
-
-type Evolve<'DomainEvent> = 
-    EffectiveEvents<'DomainEvent>
-     -> 'DomainEvent
-     -> EffectiveEvents<'DomainEvent>
-
-type Interpret<'DomainCommand,'DomainEvent> = 
-    'DomainCommand 
-     -> EffectiveEvents<'DomainEvent>
-     -> Result<unit,string> * 'DomainEvent list
-
-type Execute<'EntityId,'DomainCommand> =
-    'EntityId
-     -> 'DomainCommand 
-     -> AsyncResult<unit,string>
-
-type Projection<'DomainEvent,'View> =
-    ObservationDate
-     -> EffectiveEvents<'DomainEvent>
-     -> 'View
-
-type Query<'EntityId,'DomainEvent,'View> = 
-    'EntityId
-     -> ObservationDate 
-     -> Projection<'DomainEvent,'View>
-     -> AsyncResult<'View,string>
-
-type Aggregate<'DomainCommand,'DomainEvent,'DomainState> =
+type Aggregate =
```

Comment:
What about just doing a normal actual `type` ?

```
type Aggregate(name) =
    member __.Entity = name
    member __.Initial = initial
    member __.EvolveDomain: LeaseState -> LeaseEvent -> LeaseState = evolveDomain?
```

(NB I'm not sure what this type is even for (and I'm not sure exposing all those things is necessarily useful, seems like an ISP violation to me) - my one and only point is that I dont get the purpose of putting functions in a record - I may be underinformed and/or missing something)

Reply:
I had seen this done somewhere and thought it looked nice so just adopted it. I do think the classes are cleaner for exposing only the functions you need. will refactor on the next upate